app/models/dashboards/vacancies.rb

```diff
@@ -0,0 +1,88 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/boston-cas/blob/master/LICENSE.md
+###
+#
+class Dashboards::Vacancies < Dashboards::Base
+  include ArelHelper
+
+  def initialize(start_date:, end_date:, match_route_name:, program_types:)
+    @start_date = start_date
+    @end_date = end_date
+    @match_route_name = match_route_name
+    @program_types = program_types
+  end
+
+  def in_progress
+    @in_progress ||= reporting_scope.
+      merge(Reporting::Decisions.in_progress).
+      distinct.
+      pluck(r_d_t[:cas_client_id]).
+      map do |id|
+        { id: id }
+      end
+  end
+
+  def vacancy_results
+    hash = reporting_scope.
+      merge(Reporting::Decisions.terminated).
+      distinct.
+      pluck(r_d_t[:terminal_status], :cas_client_id).group_by(&:first)
+    hash.merge(hash) do |_key, ids|
+      ids.map do |_status, id|
+        { id: id }
+      end
+    end
+  end
+
+  def vacancies_filled(start_date:, end_date:)
+    hash = reporting_scope.
+      merge(Reporting::Decisions.success).
+      merge(Reporting::Decisions.current_step).
+      merge(Reporting::Decisions.ended_between(start_date: start_date, end_date: end_date)).
```

Comment:
Any reason not to `merge(Reporting::Decisions.success.current_step.ended_between(start_date: start_date, end_date: end_date))`?